main.rb

```diff
@@ -0,0 +1,141 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for item in self
+      yield item
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    counter = 0
+    my_each do |item|
+      yield item, counter
+      counter += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected = []
+    my_each { |item| selected.push(item) if yield(item) }
+    selected
+  end
+
+  def my_all?(to_check = nil)
+    return true if !to_check and length
+
+    my_each_with_index do |item, index|
+      next unless self[index + 1]
+
+      return false unless item == self[index + 1]
+    end
+    true
+  end
+
+  def my_any?(to_check = nil)
+    return true if !to_check and length
+
+    my_each do |item|
+      return true if item == to_check
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    my_each do |item|
+      return false if yield(item)
+    end
+    true
+  end
+
+  def my_count
+    counter = 0
+
+    my_each do
+      counter += 1
+    end
+
+    counter
+  end
+
+  def my_map(&my_proc)
+    return to_enum(:my_map) unless block_given?
+
+    map_list = []
+
+    my_each { |item| map_list.push(my_proc.call(item)) }
+    map_list
+  end
+
+  def my_inject(initial = nil)
+    total = initial || self[0]
+    counter = initial ? 0 : 1
+
+    my_each do
+      total = yield(total, self[counter]) if self[counter]
+      counter += 1
+    end
+    total
+  end
+end
+
```

Comment:
### `my_inject?` should be identical and returns the same thing as ruby's [`inject?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject).